app/assets/javascripts/models/beta/story.js

```diff
@@ -330,27 +330,38 @@ export const mergeWithFetchedStories = (
   fetchedStories,
   pastStoryIds
 ) => {
-  const storyNotSaved = currentStories.filter(isNew);
-
-  const editedStories = fetchedStories.map((fetchedStory) => {
-    const existingStory = currentStories.find(
-      (story) => story.id === fetchedStory.id
-    );
-    if (existingStory && !existingStory.collapsed) {
-      return {
-        ...fetchedStory,
+  const mergedStories = { ...fetchedStories };
+
+  if (Object.values(currentStories).length === 0) {
+    return mergedStories;
+  }
+
+  if (currentStories.stories[null]) {
+    mergedStories.stories[null] = currentStories.stories[null];
```

Comment:
This will cause mutability in the `fetchedStories`. Using the spread operator in line `333` won´t clone the whole object, it will just copy the references of the attributes of `fetchedStories` to a new object, but `mergedStories.stories` and `fetchedStories.stories` are the exact same object in memory, so modifying one of them will modify the other as well.